### PR TITLE
Implement MailToil and mailsend functionality

### LINKED ISSUE #2: Implement MailToil/mailsend functionality

### LINKED ISSUE #3: Use Victoria encryption providers to encrypt/decrypt sensitive data in MailToil config

### PR DESCRIPTION
**Pull request by Figglewatts**
- Closes #3 
- Closes #2 
- Please note, a lot of this code is ported straight over from the old scripts with minor adjustments to work with Victoria. These scripts were written before a lot of our processes in terms of code quality and style were set in stone so there may be things missing that normally would be there. This was left as-is so as not to affect functionality. 
These scripts have been battle-tested in on-call scenarios and while they are a bit rough around the edges, they work. Bringing them up to our more modern standards is a nontrivial chunk of work and will have to be marked as tech debt.

### CODE REVIEW

**Comment by Figglewatts** [comment]
The 3.6 build is failing because we use Python features that were added in 3.7. 3.6 support will be removed by #5

File: victoria_email/core/blob_storage.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
+from io import BytesIO
+import json
+import logging
+
+from azure.common import AzureMissingResourceHttpError
+from azure.storage.blob import BlobServiceClient
+import azure.storage.blob
+
+# sometimes it's in different places...
```
Comment (Alex-Potter-Dixon) [inline victoria_email/core/blob_storage.py:9]: Can you be more specific here i.e. The mime message in the Azure blob storage is sometimes located in different folder locations.

File: victoria_email/core/blob_storage.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    Args:
+        transaction_id (str): The transaction ID to use to get the message.
+        blob_service (BlockBlobService): The service to use to get the message.
+
+    Returns:
+        str: The MIME message.
+    """
+    for blob_name in MIME_BLOB_NAMES:
+        try:
+            print(transaction_id, blob_name)
```
Comment (Alex-Potter-Dixon) [inline victoria_email/core/blob_storage.py:42]: Print or Logging?

File: victoria_email/core/blob_storage.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            blob_content.write(blob.readall())
+            blob_content = blob_content.getvalue().decode('utf-8')
+
+            if blob_name.endswith(".json"):
+                mime_json = json.loads(blob_content)
+                return mime_json["receivedMimeMessage"]
+            else:
+                return blob_content
+        except AzureMissingResourceHttpError:
+            # this is pure filth, but some of the blobs are in messed up formats
```
Comment (Alex-Potter-Dixon) [inline victoria_email/core/blob_storage.py:56]: Could you add some logging here just so it's clear if something has gone wrong?

File: victoria_email/core/mail_reconstruction.py [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+from email.mime.message import MIMEMessage
+import logging
+from os.path import join
+import re
+from typing import List, Union, Tuple, Iterator
+
+ANONYMISED_SUFFIX = "_anon"
+"""The suffix to put on a file that has been anonymised."""
+
+EMAIL_PATTERN = "<(.*?)>"
```
Comment (Alex-Potter-Dixon) [inline victoria_email/core/mail_reconstruction.py:14]: I know it's obvious but still prefer a comment for a global

File: victoria_email/core/mail_reconstruction.py [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+    This is useful for if you want to open a reconstructed email in outlook.
+
+    Args:
+        output_dir (str): The directory to write the file to.
+        transaction_id (str): The transaction ID of the email.
+        message (MIMEMessage): The parsed MIME message to write.
+        suffix (str): The suffix to add to the filename.
+    """
+    filename = transaction_id + suffix + ".eml"
+    logging.info(f"    Writing message '{filename}'")
```
Comment (Alex-Potter-Dixon) [inline victoria_email/core/mail_reconstruction.py:54]: Why whitespace?

File: victoria_email/recover_mail.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+from os.path import join
+from typing import List
+import smtplib
+
+from .core import config, service_bus, blob_storage, mail_reconstruction
+from .schemas import EmailConfig
+
+
+def recover(cfg: config.MailToilConfig, cluster: str, input_file: str,
+            smtp_addr: str, plugin_cfg: EmailConfig) -> None:
```
Comment (Alex-Potter-Dixon) [inline victoria_email/recover_mail.py:21]: method docs missing

File: victoria_email/core/service_bus.py [hunk 7]
```diff
@@ -0,0 +1,115 @@
+import logging
```
Comment (Alex-Potter-Dixon) [inline victoria_email/core/service_bus.py:1]: module doc?

File: victoria_email/core/mail_reconstruction.py (hunk 5 above)
Comment (Figglewatts) [reply victoria_email/core/mail_reconstruction.py:54]: So it's easier to read in the output to the terminal, groups them under a heading in a sense.

File: victoria_email/core/blob_storage.py (hunk 3 above)
Comment (Figglewatts) [reply victoria_email/core/blob_storage.py:56]: Logging not needed, it's just fall through to the next location in blob storage. If it doesn't find any it breaks out of the for loop and logs that the transaction ID could not be found in blob storage.

**Pull request merged**